src/models/dashboard-items.js

```diff
@@ -256,6 +283,41 @@ export function buildDashboardItems(
         .flatten()
         .value();
 
+    const tablesByAntigenAndDose = _(antigensMeta)
+        .flatMap(antigen => {
+            const doses = disaggregationMetadata.doses(antigen);
+            if (!doses)
+                return getTables({
+                    tables: tablesByAntigenAndDoseMetadata,
+                    antigen,
+                    elements,
+                    organisationUnits: organisationUnitsMetadata,
+                    itemsMetadata,
+                    disaggregationMetadata,
+                    legendsMetadata,
+                });
+            const dosesCategoryId = doses.categoryId;
+            const dosesForTables = doses.elements.map(d => ({
+                categoryId: dosesCategoryId,
+                doseId: d.id,
+                name: d.name,
+            }));
+            return dosesForTables.map(dft =>
+                getTables({
+                    tables: tablesByAntigenAndDoseMetadata,
+                    antigen,
+                    elements,
+                    organisationUnits: organisationUnitsMetadata,
+                    itemsMetadata,
+                    disaggregationMetadata,
+                    legendsMetadata,
+                    doseMetadata: dft,
+                })
+            );
+        })
+        .flatten()
```

Comment:
The previous block returns either `Table[]` or Table[][]`, usually they should return the same tyope. This can be fixed with `return _.flatMap(dosesForTables, dft =>`. Then this flatten() can be removed.